lib/logstash/codecs/cloudwatch_logs.rb

```diff
@@ -25,11 +31,13 @@ def decompress(data)
   end
 
   def parse(json, &block)
-    base = json.reject { |k,_| k == "logEvents" }.freeze
-    events = json["logEvents"]
+    events = json.delete("logEvents")
+    json.freeze
 
     events.each do |event|
-      yield LogStash::Event.new(base.merge(event))
+      epochmillis = event.delete("timestamp").to_i
+      event[LogStash::Event::TIMESTAMP] = LogStash::Timestamp.at(epochmillis / 1000, (epochmillis % 1000) * 1000)
```

Comment:
Got it. My knowledge of the awslogs agent is limited. My use case is actually Docker's [native awslogs driver](https://docs.docker.com/engine/admin/logging/awslogs/), which precludes setting timestamp at ingest. So, post-processing with the `date` filter is not exclusively for cases of CloudWatch Logs being "done wrong." Alas!

At any rate, regardless of the agent's functionality, that [PutLogEvents](http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_InputLogEvent.html#CWL-Type-InputLogEvent-timestamp) supports timestamp is reason enough make this the default. That the awslogs agent exposes a `datetime_format` option is gravy.